include/knowhere/comp/thread_pool.h

```diff
@@ -89,6 +89,20 @@ class ThreadPool {
         return pool_.numThreads();
     }
 
+    void
+    SetNumThreads(uint32_t num_threads, std::mutex& pool_mtx) {
+        if (num_threads == 0) {
+            LOG_KNOWHERE_ERROR_ << "set number of threads can not be 0";
+            return;
+        } else {
+            std::lock_guard<std::mutex> lock(pool_mtx);
+            // setNumThreads() adjust the relevant variables instead of changing the number of threads directly;
+            // If numThreads < active threads, reduce number of running threads.
+            pool_.setNumThreads(num_threads);
```

Comment:
Is this Idempotent?

Reply:
A std::unique_lock inside setNumThreads function. I remove this lock.